### ISSUE TITLE
Bugs with snaps

### ISSUE THREAD

**Issue by jtsang586** [vegaprotocol/frontend-monorepo#4641]
## Description

Related to #4510 

This ticket includes a collection of issues found whilst testing the early build of vega wallet snaps. 

# 1. Unable to switch network

### Steps to Reproduce

1. Install version of metamask that allows snaps
2. Open up Governance site 
3. Click connect vega wallet and go through flow connect metamask with vega snaps
4. Error shown for incorrect network, clicking try again does not solve the issue

### Expected behaviour

If my wallet was set to a different network there should be an option to change the network to the correct one - or if that feature is not yet available prompt me to connect wallet using other connecters e.g. cli/browser extension.

![Screenshot 2023-08-29 at 11 43 53](https://github.com/vegaprotocol/frontend-monorepo/assets/30622993/b1480c93-4d5a-485f-ab4e-8a4a00d1b3dc)

# 2. Unable to import wallet

When connecting to snaps there is no prompt to import wallet. Instead it creates a new wallet to be used by snaps, this creates a separate issue where the mnemonic is not given to the user so they have no way of importing that wallet anywhere.

# 3. Unable to switch to CLI wallet

### Steps to Reproduce

1. Click connect vega wallet and go through flow connect metamask with vega snaps
2. Disconnect snaps wallet
3. Start up service for cli wallet
4. The button to connect cli wallet stays greyed out

**Comment by jtsang586** [comment]
Currently blocked by https://github.com/vegaprotocol/vega/issues/9206

**Comment by asiaznik** [comment]
1 - It seems that some of the fairground nodes are not resulting with chain id
2 - not related to frontend
3 - can't reproduce
